ORM/ORMFundamentals/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.EntityFrameworkCore;
+using ORMFundamentals.Data;
+using System.Linq.Expressions;
+
+namespace ORMFundamentals.Repositories
+{
+    public abstract class BaseRepository<T>(AppDbContext context) : IBaseRepository<T> where T : class
+    {
+        protected readonly AppDbContext _context = context;
+
+        public async Task<IEnumerable<T>> GetAll(bool trackChanges = true)
+        {
+            return trackChanges ?
+                await _context.Set<T>().ToListAsync() :
```

Comment:
`Set<T>` repeats too many times, how about storing this field in class level? You can use example like [this](https://github.com/dotnet-fizzyy/Diploma/blob/V2/WebAPI/WebAPI/Infrastructure/Postgres/Repository/BaseCrudRepository.cs#L19), it reduces duplication